brainbox/modeling/utils.py

```diff
@@ -121,9 +125,10 @@ def _get_best_new_feature(self, mask, cells):
             fitfeatures = self.features[candidate_mask]
             feat_idx = np.hstack([self.design.covar[feat]['dmcol_idx'] for feat in fitfeatures])
             mdm = self.design[np.ix_(self.train, feat_idx)]
+            mdm_test = self.design[np.ix_(self.test, feat_idx)]
             coefs, intercepts = self.model._fit(mdm, my, cells=cells)
             for i, cell in enumerate(cells):
                 scores.at[cell, feature_idx] = self.model._scorer(coefs.loc[cell],
                                                                   intercepts.loc[cell],
-                                                                  mdm, my[:, i])
-        return scores.idxmax(axis=1), scores.max(axis=1)
```

Comment:
The reason I've hesitated in changing this so far is because technically if you do this you will have a test/train leak. [See this issue on the new repo](https://github.com/berkgercek/neurencoding/issues/4), but essentially you're informing your results with the test set which should be strictly for evaluation.

The proper thing to do would be to have a second line here where the model is evaluated yet again on the test set, and the method returns both the train and test scores. It's high up on my to-do list, likely this week

Reply:
Thank you Berk! The changes you've made look great :). "full_scores" is exactly what I want, and I agree with your train/test evaluation issue.

Would it be possible to have these changes in the "develop" branch of ibllib?

Reply:
It would be preferable if you switch your code over to use the neurencoding package; the interface hasn't changed at all (still uses the DesignMatrix and LinearGLM/PoissonGLM classes), you just need to ctrl F -> replace all of the instances of brainbox.modeling. I'm supporting that one and it has some bugfixes, and should be easy to install with conda develop.

Reply:
Also I fixed the test-train issue in the neurencoding repo! The sfs object will now have a `.test_scores_` and `.train_scores_` method, as well as a flag `full_scores` to choose whether you save all the intermediate steps too

Reply:
No, they just prevented some of the helper functions (`.add_boxcar` and `.add_covariate_raw`) from working at all. If you didn't run into them you're fine I think

Reply:
Hey Hyun, let's move the discussion to the [issue here on the other repo](https://github.com/berkgercek/neurencoding/issues/4). I'm not sure which file you're talking about, in 197 and 198 of the new repo on SequentialSelector there is .loc[cell, i]?